formats/src/main/kotlin/com/jakewharton/diffuse/format/Jar.kt

```diff
@@ -7,6 +7,7 @@ import com.jakewharton.diffuse.io.Input
 
 class Jar private constructor(
   override val filename: String?,
+  val bytecodeVersion: Short?,
```

Comment:
Unfortunately I think we're going to drown in nuance here. Jars don't have bytecode versions, classes do. And multiple classes within a jar can have different versions. You can conditionally classload classes based on capabilities and those classes could be compiled for two different versions of Java. There's also the multi-version jar standard which puts classes of different versions into `META-INF/versions/N/`. The ordering of the classes in the jar (read: zip) could be effectively random so taking the first class may not be indicative of what the majority of classes were compiled with.

I'm happy to discuss strategies for surfacing this information, but as-is I think this is too simple of an implementation. Note that it probably works for 99.99% of jars, but when it fails it fails in a spectacularly wrong way.

Reply:
Now I count all the byte code versions in a Jar and filter the most one.